<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN">
<html>
<head>
<meta name="copyright" content="2000,  Bradley Kjell" />
<meta name="author" content="Bradley Kjell kjell@ieee.org " />
<meta name="translation" content="Copyright 2004 Deutsche &Uuml;bersetzung Heinrich Gailer heinrich@gailer-net.de" />
<meta name="robots" content="index,follow" />
<title>try{} und catch{}</title>
<link rel="stylesheet" type="text/css" href="../CAIstyle.css" />
</head>

<body>
<!-- generated by CSSmaker.java, version of 10/12/2004-->
<!-- from source javaChap80.frames on Thu Sep 11 15:46:28 CEST 2008 -->

<!-- ANSWER AREA -->


<div class="answer">
<h3>Eine gute Antwort w&auml;re:</h3>
<p>
Da das eine Ausnahme ist, kann das Programm sie behandlen
(mit dem richtigen Code.)
</p>


</div>

<!-- LESSON AREA -->
<div class="lesson">
<h1><code><strong>try{}</strong></code> und <code><strong>catch{}</strong></code></h1>
<p>
Um eine Ausnahme aufzufangen:
</p>

<ol>
<li>Stellen Sie Code, der eine Ausnahme ausl&ouml;sen k&ouml;nnte, innerhalb eines <code><strong>try{}</strong></code>-Blocks.</li>
<li>Stellen Sie Code, der eine Ausnahme behandelt innerhalb eines <code><strong>catch{}</strong></code>-Blocks.</li>
</ol>

<p>
Hier ist das Beispielprogramm mit zus&auml;tzlichen Code das zu tun:
</p>

<pre class="program">
import java.lang.* ;
import java.io.* ;
public class Quadrat
{
  public static void main ( String[] a ) throws IOException
  {
    BufferedReader stdin =
        new BufferedReader ( new InputStreamReader( System.in ) );
    String inData;
    int    zahl ;
    System.out.println("Eine Zahl eingeben:");
    inData = stdin.readLine();

    <span class="blue">try
    {</span>
      zahl    = Integer.parseInt( inData );
      System.out.println("Das Quadrat von " + inData + " ist " + zahl*zahl );
    <span class="blue">}</span>

    <span class="blue">catch (NumberFormatException ex )
    {</span>
      System.out.println("Sie haben falsche Daten eingegeben." );
      System.out.println("Starten Sie das Programm erneut." );
    <span class="blue">}</span>
    System.out.println("Good-bye" );
  }
}
</pre>


<p>
Wenn eine Anweisung inerhalb des <code>try{}</code>-Blocks eine
<code>NumberFormatException</code> ausl&ouml;st,
wird sofort der <code>catch{}</code>-Block ausgef&uuml;hrt.
Die verbleibenden Anweisungen in dem <code>try{}</code>-Block werden ausgelassen.

Die Referenzvariable <code>ex</code> verweist auf das
<code>Exception</code>-Objekt.
(Aber, dieses Beispiel tut nichts damit.)
</p>

<p>
Nachdem der <code>catch{}</code>-Block ausgef&uuml;hrt ist,
wird die Ausf&uuml;hrung mit der Anweisung fortgesetzt, die dem
<code>catch{}</code>-Block folgt.
(Die Ausf&uuml;hrung kehrt nicht zum <code>try{}</code>-Block zur&uuml;ck.)
</p>


</div>

<!-- QUESTION AREA -->
<div class="question">
<h3> FRAGE 5:</h3>
<a href="ch80_6.html">
<img src="../nextBut.gif" hspace="5" align="right" border="0"/></a>

<p>
Was gibt das Programm bei diesem Input aus?
</p>

<table border bordercolor="blue" bgcolor="white" width="80%">
<tr bgcolor="lightblue"><th>Input 1:</th><th>Input 2:</th></tr>
<tr align="center"><td>Unsinn</td> <td>12</td></tr>
</table>

<br clear="all" />

</div>
<div>
<a title="Zurück zum Inhaltsverzeichnis gehen." href="../../java-toc.html#80">Inhaltsverzeichnis</a>
</div>

</body>
</html>
